fw/t/unit/test_http_parser.c

```diff
@@ -146,10 +147,14 @@ split_and_parse_n(unsigned char *str, int type, size_t len, size_t chunk_size)
 			chunk_size = len - pos;
 		TEST_DBG3("split: len=%zu pos=%zu\n",
 			  len, pos);
+		chunk = kmalloc(chunk_size, GFP_KERNEL);
+		BUG_ON(chunk == NULL);
+		memcpy(chunk, str + pos, chunk_size);
 		if (type == FUZZ_REQ)
-			r = tfw_http_parse_req(req, str + pos, chunk_size, &parsed);
+			r = tfw_http_parse_req(req, chunk, chunk_size, &parsed);
 		else
-			r = tfw_http_parse_resp(resp, str + pos, chunk_size, &parsed);
+			r = tfw_http_parse_resp(resp, chunk, chunk_size, &parsed);
+		kfree(chunk);
```

Comment:
That's a use-after-free error. `TfwHttpReq`/`TfwHttpResp` store references into those chunks so you cannot deallocate them before the `req`/`resp` is released. After commenting out the `kfree(chunk)` line I got `tfw_test: finish - all passed`.